src/csa_processvariable.cpp

```diff
@@ -1741,43 +1348,59 @@ UA_StatusCode ua_processvariable::addPVChildNodes(UA_NodeId pvNodeId, string bas
   if(addResult == UA_STATUSCODE_GOOD) {
     UA_NodeId descVariable = UA_NODEID_NUMERIC(CSA_NSID, CSA_NSID_VARIABLE_DESC);
     NODE_PAIR_PUSH(this->ownedNodes, descVariable, createdNodeId);
+    UA_NodeId_copy(&createdNodeId, &mapElemDesc.concreteNodeId);
+    map.push_back(mapElemDesc);
   }
   else
     return addResult;
 
   UA_NodeId_clear(&createdNodeId);
   createdNodeId = UA_NODEID_NULL;
 
-  //Adding the EngineeringUnit node to the PV
+  //Adding the 'EngineeringUnit' node to the PV
+  UA_DataSource_Map_Element mapElemEU;
+  mapElemEU.typeTemplateId = UA_NODEID_NUMERIC(CSA_NSID, CSA_NSID_VARIABLE_UNIT);
+  mapElemEU.description = UA_LOCALIZEDTEXT((char*)"", (char*)"");
+  mapElemEU.dataSource.read = ua_readproxy_ua_processvariable_getEngineeringUnit;
+  mapElemEU.dataSource.write = NULL;
   UA_VariableAttributes_init(&attr);
   attr = UA_VariableAttributes_default;
   attr.displayName = UA_LOCALIZEDTEXT((char*)"en_US", (char*)"EngineeringUnit");
```

Comment:
Here you use "en_US" but for all other additional nodes you use "".
Please be consistent and add "en_US" everwhere...

Reply:
Solved.

Reply:
> Now I get quite a lot messages like: `[2023-04-27 09:58:59.111 (UTC+0200)] info/session SecureChannel 0 | Session "Administrator" | AddNode (ns=1;s=watchdog_server/Configuration/Configuration/MicroDAQ/decimationFactor/EngineeringUnit): The value is empty. But this is only allowed for BaseDataType. Create a matching default value.`
> 
> Can we set a default - like empty string and data valid?

Done for Validity.